File: 1904a_forked.cpp
/**
 * @file 1904a_forked.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Forked!(https://codeforces.com/problemset/problem/1904/A)
 * @version 0.1
 * @date 2024-03-14
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	short t, i, ans;
	int a, b, xk, yk, xq, yq;
	vector<pair<short, short>> directions = {{1, 1}, {1, -1}, {-1, 1}, {-1, -1}};
	set<pair<short, short>> posk, posq;

	cin >> t;
	while (t--)
	{
		cin >> a >> b >> xk >> yk >> xq >> yq;
		posk.clear();
		posq.clear();
		ans = 0;
		for (i = 0; i < 4; i++)
		{
			posk.insert({xk + a * directions[i].first, yk + b * directions[i].second});
			posq.insert({xq + a * directions[i].first, yq + b * directions[i].second});
		}
		for (i = 0; i < 4; i++)
		{
			posk.insert({xk + b * directions[i].first, yk + a * directions[i].second});
			posq.insert({xq + b * directions[i].first, yq + a * directions[i].second});
		}
		for (pair<short, short> posk_en : posk)
			posq.find(posk_en) != posq.end() ? ans++ : 0;
		cout << ans << '\n';
	}
	return (EXIT_SUCCESS);
}

File: 1904a_forked.py
"""
@file: 1904a_forked.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Forked!(https://codeforces.com/problemset/problem/1904/A)
@version: 1.0
@date: 2025-02-06

@copyright: Copyright (c) 2025
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(*args, sep=" ", end="\n"):
    return stdout.write(sep.join(str(x) for x in args) + end)


def main():
    for _ in range(int(input())):
        a, b = [int(x) for x in input().split()]
        xk, yk = [int(x) for x in input().split()]
        xq, yq = [int(x) for x in input().split()]
        apk, apq = set(), set()
        for dir in [(-1, -1), (-1, 1), (1, -1), (1, 1)]:
            apk.add((xk + dir[0] * a, yk + dir[1] * b))
            apk.add((xk + dir[0] * b, yk + dir[1] * a))
            apq.add((xq + dir[0] * a, yq + dir[1] * b))
            apq.add((xq + dir[0] * b, yq + dir[1] * a))
        print(len(apk & apq))


if __name__ == "__main__":
    main()
